addons/crm_livechat/views/crm_lead_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="crm_lead_view_form" model="ir.ui.view">
+        <field name="name">crm.lead.view.form.inherit.crm.livechat</field>
+        <field name="model">crm.lead</field>
+        <field name="inherit_id" ref="crm.crm_lead_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='action_schedule_meeting']" position="after">
+                <button name="action_open_livechat" type="object" class="oe_stat_button" icon="fa-comments" invisible="not from_livechat" groups="sales_team.group_sale_salesman">
```

Comment:
Just make a visible based on channel_id

Reply:
channel_id is limited to `base.group_erp_manager` group to avoid users to assign arbitrary channels to leads thus granting access to channel through ACLs. Do you have a recommendation to solve this issue?

Reply:
Do it the inverse way. Fix channel ACLs and let m2o do their job. You should use the ORM, not try to constraint it to do the inverse it was meant to be used.

I will remove the field before the freeze, then please fix channel ACLs accordingly. We can't add fields like that on models like crm.lead, it is too big to hold useless content.